cloudsim-plus-testbeds/src/main/java/org/cloudsimplus/migration/ExampleMigrationVmWhenCpuMetricIsViolated.java

```diff
@@ -0,0 +1,340 @@
+import org.cloudbus.cloudsim.datacenters.Datacenter;
+import org.cloudbus.cloudsim.datacenters.DatacenterCharacteristics;
+import org.cloudbus.cloudsim.datacenters.DatacenterCharacteristicsSimple;
+import org.cloudbus.cloudsim.datacenters.power.PowerDatacenter;
+import org.cloudbus.cloudsim.hosts.power.PowerHost;
+import org.cloudbus.cloudsim.hosts.power.PowerHostUtilizationHistory;
+import org.cloudbus.cloudsim.power.models.PowerModelLinear;
+import org.cloudbus.cloudsim.provisioners.PeProvisionerSimple;
+import org.cloudbus.cloudsim.provisioners.ResourceProvisionerSimple;
+import org.cloudbus.cloudsim.resources.Bandwidth;
+import org.cloudbus.cloudsim.resources.Pe;
+import org.cloudbus.cloudsim.resources.PeSimple;
+import org.cloudbus.cloudsim.resources.Ram;
+import org.cloudbus.cloudsim.schedulers.cloudlet.CloudletSchedulerDynamicWorkload;
+import org.cloudbus.cloudsim.schedulers.vm.VmSchedulerTimeShared;
+import org.cloudbus.cloudsim.selectionpolicies.power.PowerVmSelectionPolicyMinimumUtilization;
+import org.cloudbus.cloudsim.util.Log;
+import org.cloudbus.cloudsim.utilizationmodels.UtilizationModel;
+import org.cloudbus.cloudsim.utilizationmodels.UtilizationModelArithmeticProgression;
+import org.cloudbus.cloudsim.utilizationmodels.UtilizationModelFull;
+import org.cloudbus.cloudsim.vms.Vm;
+import org.cloudbus.cloudsim.vms.power.PowerVm;
+import org.cloudsimplus.sla.readJsonFile.SlaMetric;
+import org.cloudsimplus.sla.readJsonFile.SlaReader;
+import org.cloudsimplus.util.tablebuilder.CloudletsTableBuilderHelper;
+
+/**
+ * This example shows how to do a migration using CpuUtilization threshold
+ * established in the SLA Contract.
+ * When the CPU metric is violated the migration is made.
+ * 
+ * @author raysaoliveira
+ */
+public class ExampleMigrationVmWhenCpuMetricIsViolated {
+
+    private static final int SCHEDULE_TIME_TO_PROCESS_DATACENTER_EVENTS = 5;
+    private static final double DATACENTER_COST_PER_CPU = 3.0;
+    private static final double DATACENTER_COST_PER_RAM = 0.05;
+    private static final double DATACENTER_COST_PER_STORAGE = 0.001;
+    private static final double DATACENTER_COST_PER_BW = 0.0;
+
+    private static final int HOST_MIPS_BY_PE = 1000;
+    private static final int HOST_NUMBER_OF_PES = 2;
+    private static final long HOST_RAM = 500000; //host memory (MB)
+    private static final long HOST_STORAGE = 1000000; //host storage
+    private static final long HOST_BW = 100000000L;
+
+    private static final int VM_MIPS = 1000;
+    private static final long VM_SIZE = 1000; //image size (MB)
+    private static final int VM_RAM = 10000; //vm memory (MB)
+    private static final long VM_BW = 100000;
+    private static final int VM_PES_NUM = 1; //number of cpus
+
+    private static final long CLOUDLET_LENGHT = 20000;
+    private static final long CLOUDLET_FILESIZE = 300;
+    private static final long CLOUDLET_OUTPUTSIZE = 300;
+    private static List<Cloudlet> list = new ArrayList<>();
```

Comment:
Only the constants and the main method must be defined as static. Remove the static keyword from the other elements and move the code inside the main method to a default constructor and instantiate the class inside the main.